### ISSUE TITLE
Importer allows for duplicate purchase keys and strings with spaces

### ISSUE THREAD

**Issue by scottbuscemi** [easydigitaldownloads/easy-digital-downloads#5189]
When using the importer, it does not heck for duplicate purchase keys or strings that have spaces in them. As a result, imported orders with key values such as "QTY 1" will  not allow customers to view their orders within the purchase history (the links to each order don't work) and the downloading of files from previous purchases doesn't work since the URL isn't being generated properly.

**Comment by scottbuscemi** [comment]
As a side-issue, I'd love if the importer allowed for importing notes... :)

**Comment by cklosowski** [comment]
Purchase keys are typically brought in as invoice IDs or something from the gateway. We should probably unique and trim these so we don't get conflicts.

**Comment by scottbuscemi** [comment]
Thanks. Should I create a separate issue for the purchase notes? There's currently no way to have notes imported.

**Comment by cklosowski** [comment]
Yeah go ahead, not sure how we're going to do it yet, but we could look into supporting it.
